deploy/openshift/deploy_che.sh

```diff
@@ -14,563 +14,376 @@
-[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [38;5;17m0[38;5;60m0[38;5;60m0[38;5;60m1[38;5;60m0[38;5;17m1[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m 
+[0m [0m [0m [0m [38;5;23m0[38;5;60m0[38;5;60m0[38;5;67m0[38;5;103m0[38;5;103m0[38;5;67m0[38;5;67m1[38;5;67m1[38;5;67m1[38;5;67m1[38;5;67m0[38;5;67m0[38;5;67m1[38;5;67m0[38;5;67m0[38;5;67m0[38;5;67m0[38;5;67m0[38;5;67m1[38;5;67m1[38;5;67m1[38;5;67m0[38;5;67m1[38;5;67m1[38;5;67m1[38;5;103m1[38;5;103m0[38;5;67m0[38;5;60m0[38;5;60m0[38;5;23m0[0m [0m [0m [0m
+[0m [0m [0m [0m [0m [0m [0m [0m [38;5;17m0[38;5;59m0[38;5;60m0[38;5;67m1[38;5;103m1[38;5;103m0[38;5;103m1[38;5;67m0[38;5;67m1[38;5;67m1[38;5;67m0[38;5;67m1[38;5;67m0[38;5;67m0[38;5;67m0[38;5;67m1[38;5;67m0[38;5;103m0[38;5;103m0[38;5;103m1[38;5;67m1[38;5;60m1[38;5;60m1[38;5;17m0[0m [0m [0m [0m [0m [0m [0m [0m
+[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [38;5;17m1[38;5;59m1[38;5;60m0[38;5;60m1[38;5;67m0[38;5;103m1[38;5;103m1[38;5;67m1[38;5;67m0[38;5;103m0[38;5;103m0[38;5;67m0[38;5;60m1[38;5;60m0[38;5;59m0[38;5;17m1[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m
+[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [38;5;17m0[38;5;60m0[38;5;60m0[38;5;60m1[38;5;60m0[38;5;17m1[0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m [0m
 [0m
 EOF
 echo
 
-# --------------------------------------------------------
-# Parse options
-# --------------------------------------------------------
+HELP="
+--help - script help menu
+--project | -p - OpenShift namespace to deploy Che (defaults to eclipse-che)
+--multiuser - Deploy che in multiuser mode
+--no-pull - IfNotPresent pull policy for Che server deployment
+--rolling - Rolling update strategy (Recreate is the default one). With Rolling strategy Che server pvc and volume aren't created
+--wait-che - Track Che deployment progress until pod is healthy
+--debug - Deploy Che in a debug mode, create and expose debug route
+--image-che= - Override default Che image. Example --image-che=org/repo:tag. Tag is mandatory!
+===================================
+ENV vars: this script automatically detect envs vars beginning with "CHE_" and passes them to Che deployments:
+CHE_IMAGE_REPO - Che server Docker image, defaults to "eclipse-che-server"
+CHE_IMAGE_TAG - Set che-server image tag, defaults to "nightly"
+CHE_INFRA_KUBERNETES_PROJECT - namespace for workspace objects (defaults to eclipse-che). A separate ws namespace can be used only if username/password or token is provided
```

Comment:
`CHE_INFRA_KUBERNETES_NAMESPACE` is used for Kubernetes Infra.
`CHE_INFRA_OPENSHIFT_PROJECT` is used for OpenShift Infra.